src/modern/lib/download.ts

```
@@ -1,8 +1,19 @@
+/**
+ * Sanitizes a filename by replacing illegal characters and path traversal sequences.
+ * Replaces illegal OS characters (/ \ : * ? " < > |) and control characters with underscores.
+ * Replaces path traversal sequences (..) with double underscores (__).
+ */
+export function sanitizeFilename(filename: string): string {
+  return filename
+    .replace(/\.\./g, '__') // Replace path traversal
+    .replace(/[<>:"\/\\|?*\x00-\x1F]/g, '_'); // Replace illegal OS and control characters
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00, \x1f.

(no-control-regex)